packages/core/src/sync/index.ts

```diff
@@ -566,180 +579,241 @@ export const createSync = async (args: CreateSyncParameters): Promise<Sync> => {
-          while (true) {
-            if (isKilled) return;
-            if (from === to) break;
-            const { events, cursor } = await args.syncStore.getEvents({
-              filters: args.sources.map(({ filter }) => filter),
-              from,
-              to,
-              limit: args.common.options.syncEventsQuerySize,
-            });
+        const blockWithEventData = {
+          block: event.block,
+          filters: event.filters,
+          logs: event.logs,
+          factoryLogs: event.factoryLogs,
+          callTraces: event.callTraces,
+          transactions: event.transactions,
+          transactionReceipts: event.transactionReceipts,
+        };
 
-            for (const network of args.networks) {
-              updateRealtimeStatus({ checkpoint: cursor, network });
-            }
-            args
-              .onRealtimeEvent({ type: "block", checkpoint: to, events })
-              .then(() => {
-                if (events.length > 0 && isKilled === false) {
-                  args.common.logger.info({
-                    service: "app",
-                    msg: `Indexed ${events.length} events`,
-                  });
-                }
-              });
+        unfinalizedEventData.push(blockWithEventData);
 
-            from = cursor;
+        unindexedEvents.push(
+          ...buildEvents({
+            sources: args.sources.filter(
+              ({ filter }) => filter.chainId === network.chainId,
+            ),
+            blockWithEventData,
+            finalizedChildAddresses: realtimeSync.finalizedChildAddresses,
+            unfinalizedChildAddresses: realtimeSync.unfinalizedChildAddresses,
+          }),
+        );
+
+        if (to > from) {
+          for (const network of args.networks) {
+            updateRealtimeStatus({ checkpoint: to, network });
           }
 
-          break;
-        }
-        /**
-         * Handle a new block being finalized.
-         */
-        case "finalize": {
-          // Newly finalized range
-          const interval = [
-            hexToNumber(syncProgress.finalized.number),
-            hexToNumber(event.block.number),
-          ] satisfies Interval;
-
-          // Update local sync, record checkpoint before and after
-          const prev = getOmnichainCheckpoint("finalized")!;
-          syncProgress.finalized = event.block;
-          const checkpoint = getOmnichainCheckpoint("finalized")!;
-
-          syncProgress.finalized = event.block;
-
-          // Insert an interval for the newly finalized range.
-          await Promise.all(
-            args.sources
-              .filter(({ filter }) => filter.chainId === network.chainId)
-              .map(({ filter }) =>
-                args.syncStore.insertInterval({ filter, interval }),
-              ),
+          const events: RawEvent[] = unindexedEvents.filter(
+            ({ checkpoint }) => checkpoint <= to,
+          );
+          unindexedEvents = unindexedEvents.filter(
```

Comment:
I find it a bit confusing that this is called `unindexedEvents`, because this service does not actually do the indexing. Am I correct that this is the list of events (across all chains) that the sync service is holding onto, but can't pass up to the runtime because there is another network with a lagging checkpoint? Essentially a buffer?

Reply:
Yes, exactly. Is `eventBuffer` more clear?

Reply:
Maybe `bufferedEvents` ?